modules/ci_environment/manifests/transition_logs/s3_backups.pp

```diff
@@ -0,0 +1,46 @@
+# class ci_environment::transition_logs::s3_backups
+#
+# [*directory*]
+#   Defines the directory to backup
+#
+# [*hour*]
+#   Defines the hour to run the cron job
+#
+# [*minute*]
+#   Defines the minute to run the cron job
+#
+# [*retention*]
+#   Defines the number of full backups to retain
+#
+# [*s3_bucket*]
+#   Defines the AWS S3 bucket destination for backups
+#
+# [*s3_aws_id*]
+#   Defines the AWS access id with permission to write to 
+#   the S3 bucket
+#
+# [*s3_aws_key*]
+#   Defines the AWS access secret key for the defined aws access id
+
+class ci_environment::transition_logs::s3_backups(
+  $directory = '/srv/logs/log-1',
+  $hour = 4,
+  $minute = 15,
+  $retention = 30,
```

Comment:
I think the change to `remove_all_but_n_full` means that it should probably be 4.